src/webserver/routes/api/users.ts

```diff
@@ -356,4 +357,49 @@ router.delete(
 	},
 );
 
+router.post(
+	'/users/sync',
+	async (req, res, next) =>
+		request(req, res, next, {
+			authentication: true,
+			roles: ['ADMIN'],
+		}),
+	async (req, res) => {
+		try {
+			if (!requestHasUser(req)) return;
+
+			if (
+				!req.user.tokens.scopes.includes(
+					'https://www.googleapis.com/auth/admin.directory.user.readonly',
+				)
+			)
+				return res
+					.status(403)
+					.send('You have not authorized the required scope.');
+
+			new AdminClient()
+				.startSync(req.user)
+				.then(x => {
+					res.status(200).send();
+				})
+				.catch(e => {
+					res.status(500).send(e);
+				});
```

Comment:
If we `catch` here, what is the point of the `catch` below? Wouldn't it be better to just use the code below and store errors with the sync?

Reply:
The error gets parsed in `admin.ts` so it should be explained in the response. So I'll just remove the outer try/catch and keep this.